Задание 1. Скачайте с сайта датасет любого размера. Определите, какому фильму было выставлено больше всего оценок 5.0.

In [1]:
import pandas as pd

df = pd.read_csv('ml-latest-small/ratings.csv')
df.head()

userId  movieId  rating  timestamp
0       1        1     4.0  964982703
1       1        3     4.0  964981247
2       1        6     4.0  964982224
3       1       47     5.0  964983815
4       1       50     5.0  964982931

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100836 entries, 0 to 100835
Data columns (total 4 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   userId     100836 non-null  int64  
 1   movieId    100836 non-null  int64  
 2   rating     100836 non-null  float64
 3   timestamp  100836 non-null  int64  
dtypes: float64(1), int64(3)
memory usage: 3.1 MB


In [3]:
df[df['rating'] == 5.0][['movieId']].value_counts().head()

movieId
318        153
296        123
356        116
2571       109
260        104
dtype: int64

Больше всего было у фильма под ID 318

Задание 2
По данным файла power.csv посчитайте суммарное потребление стран Прибалтики (Латвия, Литва и Эстония) категорий 4, 12 и 21 за период с 2005 по 2010 год. Не учитывайте в расчётах отрицательные значения quantity.

In [4]:
df = pd.read_csv('power.csv')

In [5]:
df.head()

country  year  quantity  category
0  Austria  1996       5.0         1
1  Austria  1995      17.0         1
2  Belgium  2014       0.0         1
3  Belgium  2013       0.0         1
4  Belgium  2012      35.0         1

In [6]:
df[df['country'].str.contains('lit', case=False)]['country'].unique()

array(['Lithuania'], dtype=object)

In [7]:
df = df[df['quantity'] >= 0][(df['category'] == 4) | (df['category'] == 12) | (df['category'] == 21)]
df = df[df['year'].between(2005, 2010)]
print('Latvia', df['quantity'][(df['country'] == 'Latvia')].sum())
print('Estonia', df['quantity'][(df['country'] == 'Estonia')].sum())
print('Lithuania', df['quantity'][(df['country'] == 'Lithuania')].sum())

Latvia 54844.0
Estonia 63440.0
Lithuania 122296.0


C:\Users\baakh\AppData\Local\Temp\ipykernel_11416\3034114418.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df = df[df['quantity'] >= 0][(df['category'] == 4) | (df['category'] == 12) | (df['category'] == 21)]


Задание 3
Выберите страницу любого сайта с табличными данными. Импортируйте таблицы в pandas DataFrame. Вы можете взять любые страницы.

In [8]:
df = pd.read_html('https://fortrader.org/quotes')

In [9]:
len(df)

8

In [10]:
df[0]

Название   Цена  Изменение  Изменение(%)  Обновлено
0  EURRUB Курс евро к рублю сегодня  73.64      -0.31         -0.42        NaN
1       USDRUB Курс рубля к доллару  68.63      -0.36         -0.52        NaN

In [11]:
df[2]

Название   Цена  Изменение  Изменение(%)  Обновлено
0       Золото Цена золота на бирже  1 845       3.00          0.00        NaN
1  Серебро Цена на серебро на бирже  23.98       0.37          1.58        NaN
2         Палладий Цена на Палладий  1 790      16.00          1.00        NaN
3           Платина Цены на платину  1 105      14.00          1.00        NaN
4           Медь Цена меди на бирже   4.12       0.04          1.10        NaN